contrib/go/src/python/pants/contrib/go/tasks/go_test.py

```diff
@@ -26,28 +33,100 @@ class GoTest(GoWorkspaceTask):
   @classmethod
   def register_options(cls, register):
     super(GoTest, cls).register_options(register)
-    register('--build-and-test-flags', default='',
-             fingerprint=True,
+    register('--build-and-test-flags', type=str, default='', fingerprint=True,
+             removal_version='1.17.0.dev0',
+             removal_hint='Use --shlexed-build-and-test-flags instead!',
              help='Flags to pass in to `go test` tool.')
+    # TODO: make a shlexed flags option type!
+    register('--shlexed-build-and-test-flags', type=list, member_type=str, fingerprint=True,
```

Comment:
Would it be accurate to just call this just `--go-flags`? It's already on a test task, and presumably "build" is implicit.

Reply:
From `go help test`, I'm seeing e.g.:
> In addition to the build flags, the flags handled by 'go test' itself are:

And it seems to be pretty specific about which flags are accepted by the binary (specifically for building and testing). I suspect the original option name was trying to be specific about that, but I do *not* have enough familiarity with go to know. `--shlexed-build-and-test-flags` is a ridiculous name, but I was thinking we would drop the `shlexed` part once the original option completes the deprecation cycle.

Reply:
Ok. Let's just remember to do the second deprecation cycle.

Reply:
I've added specific instructions on these sequential deprecation cycles in the `removal_hint` of the `--build-and-test-flags` argument!